tutorials/how-to-setup-applesilicon-server-with-terraform-ansible/index.mdx

```diff
@@ -0,0 +1,229 @@
+
+```terraform
+#resources.tf
+resource "scaleway_vpc" "vpc01" {
+    name = "MyAwesomeVPC"
+}
+
+resource "scaleway_vpc_private_network" "pn01" {
+    name = "MyAwesomePN"
+    vpc_id = scaleway_vpc.vpc01.id
+}
+
+resource "scaleway_apple_silicon_server" "server" {
+    name = "MyAwesomeServer"
+    type = "M2-M"
+    zone = "fr-par-1"
+    enable_vpc = true
+    private_network {
+          id = scaleway_vpc_private_network.pn01.id
+    }
+}
+```
+8. Apply the Configuration Update: Finally, run the following command to apply the changes and update the server configuration
+
+```shell
+terraform apply
+```
+
+This will apply the new settings, ensuring that the server is launched within the specified VPC and connected to the private network.
+
+### Using Ansible
+
+1. Install Ansible:  If you don't have Ansible installed on your system, you can easily install it using pip, the Python package manager. Run the following command:
+
+```shell
+pip install ansible
+```
+
+2. Install Scaleway Ansible Collection: Scaleway provides an official Ansible collection that enables you to interact with Scaleway resources, such as creating servers. Install this collection using the ansible-galaxy command:
+
+```shell
+ansible-galaxy collection install scaleway.scaleway
+```
+
+3. Create a Directory: Now that you have Ansible and the Scaleway collection installed, create a directory for your Ansible project. This will help you keep everything organized.
+
+Run the following commands to create and navigate into the project directory:
+
+```shell
+mkdir apple_silicon_server_ansible
+cd apple_silicon_server_ansible
+```
+
+4. Create a Playbook: In Ansible, a playbook is a YAML file that defines the tasks to automate. Create a new file for your playbook, named create_applesilicon_server.yml:
+
+```shell
+touch create_applesilicon_server.yml
+```
+
+5. Define the Playbook Content: Open the create_applesilicon_server.yml file in your preferred text editor. Add the following content to define the task of provisioning an Apple Silicon server:
+
+```ansible
+---
+- name: Create Apple Silicon Server on Scaleway
+  hosts: localhost
+  gather_facts: no
+  tasks:
+    - name: Create an Apple Silicon server
+      scaleway.scaleway.scaleway_applesilicon_server:
+        access_key: "{{ scw_access_key }}"
+        secret_key: "{{ scw_secret_key }}"
+        state: present
+        type_: "M2-M"
+        name: "my-applesilicon-server"
+        zone: "fr-par-1"
+        project_id: "{{scw_project_id}}"
+        vpc_enabled: "false"
+      register: applesilicon_server
+```
+
+6. Set Up Your Scaleway Credentials: Replace the placeholders {{ scw_access_key }}, {{ scw_secret_key }}, and {{ scw_project_id }} with your actual credentials. It’s a good practice to store these values in environment variables or an Ansible vault for added security. Alternatively, you can use a .env file or an external credentials file.
```

Comment:
Could you show a command line example about how to pass environment variables through the command line?